python/lsst/daf/butler/dimensions/_group.py

```diff
@@ -526,12 +526,24 @@ def _validate(cls, data: Any, info: pydantic.ValidationInfo) -> DimensionGroup:
         This satisfies the `pydantic.WithInfoPlainValidatorFunction` signature.
         """
         universe = pydantic_utils.get_universe_from_context(info.context)
-        return universe.conform(data)
+        return cls.from_simple(data, universe)
 
-    def _serialize(self) -> list[str]:
-        """Pydantic serializer for `DimensionGroup`.
+    @classmethod
+    def from_simple(cls, data: SerializedDimensionGroup, universe: DimensionUniverse) -> DimensionGroup:
+        """Create an instance of this class from serialized data.
 
-        This satisfies the `pydantic.PlainSerializerFunction` signature.
+        Parameters
+        ----------
+        data : `SerializedDimensionGroup`
+            Serialized data from a previous call to ``to_simple``.
+        universe : `DimensionUniverse`
+            Dimension universe in which this dimension group will be defined.
+        """
+        return universe.conform(data)
+
+    def to_simple(self) -> SerializedDimensionGroup:
+        """Convert this class to a simple data format suitable for
+        serialization.
```

Comment:
I'm not 100% on board with adding `to/from_simple` for types that are naturally Pydantic-friendly; I'd hoped to move away from them instead, as we made more things naturally Pydantic-friendly.  But I'm guessing this is to avoid using `DeferredValidation` on some other model, and that's a carve-out I can live with.

Reply:
Yeah that's why I added this -- I want to keep the top-level request/response models pydantic-context-free so that they can be validated during FastAPI's normal validation process.

In theory I could have avoided the to_simple/from_simple by adding a `SerializedDimensionGroup` as a `DeferredValidation[DimensionGroup]` instead.  But then FastAPI wouldn't be able to structurally validate it during its normal validation process, and to me having two one-line functions that explicitly declare inputs/outputs is easier to understand anyway.